converter/internal/staticconvert/internal/build/node_exporter.go

```diff
@@ -0,0 +1,119 @@
+package build
+
+import (
+	"github.com/grafana/agent/component/discovery"
+	"github.com/grafana/agent/component/prometheus/exporter/unix"
+	"github.com/grafana/agent/converter/internal/common"
+	"github.com/grafana/agent/converter/internal/prometheusconvert"
+	"github.com/grafana/agent/pkg/integrations/config"
+	"github.com/grafana/agent/pkg/integrations/node_exporter"
+)
+
+func (b *IntegrationsV1ConfigBuilder) AppendNodeExporter(nodeConfig *node_exporter.Config, commonConfig *config.Common) discovery.Exports {
+	args := ToNodeExporter(nodeConfig)
+	b.f.Body().AppendBlock(common.NewBlockWithOverride(
+		[]string{"prometheus", "exporter", "unix"},
+		"",
+		args,
+	))
+
+	return prometheusconvert.NewDiscoverExports("prometheus.exporter.unix.targets")
+}
+
+func ToNodeExporter(config *node_exporter.Config) *unix.Arguments {
```

Comment:
Yikes. But I guess it's a necessary evil - I don't see how we could make it simpler without inventing some magic conversion protocol.